### DISCUSSION TITLE
remote error: tls: user canceled with burp proxy

### DISCUSSION THREAD

**Discussion by tequreq** [projectdiscovery/katana#165] [Q&A] [answered]
I used 0.0.1 version of katana,

`katana -proxy http://127.0.0.1:8080 -u https://site.com -d 5 -jc  -v`

output : 
`[WRN] Could not request seed URL: remote error: tls: user canceled`

But trying with 'kf' param that requesting correctly. 

`./katana -proxy http://127.0.0.1:8080 -u https://site.com -d 5 -jc  -kf all -v`

output : 

```
[file] https://site.com/
[file] https://site.com/robots.txt
[file] https://site.com/m/
[file] https://site.com/mobile/
[file] https://site.com/ara?
[file] https://site.com/ara/
[file] https://site.com/admin/
[file] https://site.com/api/
[file] https://site.com/articles/
[file] https://site.com/notp/
[file] https://site.com/dev
[file] https://site.com/mesaj/
[WRN] Could not parse known files for https://site.com: could not parse sitemap: could not decode xml: XML syntax error on line 2: attribute name without = in element
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[header] https://site.com/ara/haber
[WRN] Could not request seed URL: remote error: tls: user canceled
[header] https://site.com/ara/ara/haber
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
```

I'm curious about this behavior.

**Comment by ehsandeep** [comment]
@tequreq feel free to use the discussion board for the questions.

When you use `-v` option, katana displays all warnings while trying to connect all the parsed endpoints from the response, and not all the constructed endpoints are valid or reachable, which usually results in these warnings.

Let me know if you noticed any endpoint which is valid and reachable, but katana still failed to connect and displayed under the warning section.

**Comment by tequreq** [comment]
No @ehsandeep, the domain is reachable.

https://feedly.com is live. It burp running *:8080. 

run : `./katana -proxy http://127.0.0.1:8080 -u https://feedly.com -d 5 -jc -v`

output : 
```
  __        __
  / /_____ _/ /____ ____  ___ _
 /  '_/ _  / __/ _  / _ \/ _  /
/_/\_\\_,_/\__/\_,_/_//_/\_,_/ v0.0.1

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[WRN] Could not request seed URL: remote error: tls: user canceled
```

then same configuration and network but only adding '-kf all' ; 

run : `./katana -proxy http://127.0.0.1:8080 -u https://feedly.com -d 5 -jc -v  -kf all`

output : 
```
...
[file] https://feedly.com/i/top/zombies-blogs
[file] https://feedly.com/i/top/zoo-blogs
[file] https://feedly.com/i/top/zoology-blogs
[file] https://feedly.com/i/top/zurich-blogs
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
[WRN] Could not request seed URL: remote error: tls: user canceled
...
```

I think having a tls problem but i didn't see any options for ssl or tls in katana.

**Comment by ehsandeep** [comment]
@tequreq, my guess is `feedly.com` rejecting the connection going through via burp, as it works fine and as expected if you remove the burp proxy.

```
katana -u https://feedly.com -d 5 -jc
```
in the case of `-kf all` those endpoints parsed from 1st response, so the result.

still not exactly sure.

**Reply by tequreq** [reply]
@ehsandeep, feedly.com is example of that case. Generally I used `-proxy` that I got error message.

**Reply by tequreq** [reply]
So I use with Burp.

**Reply by ehsandeep** [reply]
looks like there is some issue; I've created a ticket to improve the logging part - https://github.com/projectdiscovery/katana/issues/168 first to explore this further.

**Reply by tequreq** [reply]
@ehsandeep thanks for quickly actions.  I will have follow that issue.

**Comment by tequreq** [comment]
BTW, to give credit, the issue is discovered by @omarkurt.

**Comment by parthmalhotra** [comment, marked as answer]
Hello @tequreq : The issue is due to burp upgrading requests to HTTP/2 by default if the server supports it, Disabling the `Default to HTTP/2` in Project Options -> HTTP will fix this. 

<img width="1548" alt="Screenshot 2022-11-11 at 7 00 20 PM" src="https://user-images.githubusercontent.com/28601533/201350572-93de7e9e-f7cb-42a5-9d3f-77daeb86ba86.png">

**Reply by tequreq** [reply]
@parthmalhotra thanks bro, it issue fixed for my side. @ehsandeep fyi.

**Reply by parthmalhotra** [reply]
No problem bro :D
